### DISCUSSION TITLE
how to solve react-navigation-drawer problem?

### DISCUSSION THREAD

**Discussion by tytom2003** [react-navigation/react-navigation#9896] [Q&A]
i follow "https://reactnavigation.org/docs/nesting-navigators" to learn "Navigating to a screen in a nested navigator". When i run program, it will below the error(photo). How to solve the problem? Thank you very much.
I followed the link to install "react-navigation-drawer". I is no use Expo. 
[https://www.npmjs.com/package/react-navigation-drawer](https://www.npmjs.com/package/react-navigation-drawer)

![image](https://user-images.githubusercontent.com/25719095/131529767-df28c531-1876-4c0d-91ef-c27f55f95a0c.png)

**Comment by raajnadar** [comment]
Try clearing the metro cache `npx react-native start --reset-cache` once the terminal display the cache is cleared kill the terminal, after that `npx react-native run-android`

**Reply by tytom2003** [reply]
i use "npx react-native start --reset-cache" to clear metro cache. Then, i use "react-native run-android". it show below new error.
![image](https://user-images.githubusercontent.com/25719095/131758284-e4cd2eb0-5553-45e9-8b6e-aae78fe86b2f.png)

**Reply by tytom2003** [reply]
Below is my program.

import * as React from 'react';
import {
  View,
  Text,
  Button,
  TextInput,
  Image,
  ImageBackground,
} from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import {createDrawerNavigator} from 'react-navigation-drawer';

const Stack = createNativeStackNavigator();
const Tab = createBottomTabNavigator();
const Drawer = createDrawerNavigator();

function Home() {
  return (
    <Tab.Navigator>
      <Tab.Screen name="Feed" component={Feed} />
      <Tab.Screen name="Messages" component={Messages} />
    </Tab.Navigator>
  );
}

function Feed({navigation}) {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Profile screen</Text>
      <Button title="Go back" onPress={() => navigation.goBack()} />
    </View>
  );
}

function Messages({navigation}) {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Profile screen</Text>
      <Button title="Go back" onPress={() => navigation.goBack()} />
    </View>
  );
}

function Profile({navigation}) {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Profile screen</Text>
      <Button title="Go back" onPress={() => navigation.goBack()} />
    </View>
  );
}

function Settings({navigation}) {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Settings screen</Text>
      <Button title="Go back" onPress={() => navigation.goBack()} />
    </View>
  );
}

function Root() {
  return (
    <Drawer.Navigator>
      <Drawer.Screen name="Home" component={Home} />
      <Drawer.Screen name="Profile" component={Profile} />
      <Stack.Screen name="Settings" component={Settings} />
    </Drawer.Navigator>
  );
}

function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="Root"
          component={Root}
          options={{headerShown: false}}
        />
        <Stack.Screen name="Feed" component={Settings} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
export default App;

**Reply by raajnadar** [reply]
Please paste the content inside the `package.json` seems like you did not configure the reanimated properly, there is an extra step if you use latest version of react native reanimated

**Reply by tytom2003** [reply]
this is my package.json content.

{
  "name": "navigation",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@react-navigation/bottom-tabs": "^6.0.5",
    "@react-navigation/native": "^6.0.2",
    "@react-navigation/native-stack": "^6.1.0",
    "react": "17.0.2",
    "react-native": "0.65.1",
    "react-native-safe-area-context": "^3.3.2",
    "react-navigation-drawer": "^2.7.1"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/runtime": "^7.12.5",
    "@react-native-community/eslint-config": "^2.0.0",
    "babel-jest": "^26.6.3",
    "eslint": "7.14.0",
    "jest": "^26.6.3",
    "metro-react-native-babel-preset": "^0.66.0",
    "react-native-codegen": "^0.0.7",
    "react-test-renderer": "17.0.2"
  },
  "jest": {
    "preset": "react-native"
  }
}

thank you very muhc.

**Reply by raajnadar** [reply]
Read this https://reactnavigation.org/docs/5.x/getting-started, you missed the important dependencies for the library to work.

**Reply by tytom2003** [reply]
> Try clearing the metro cache `npx react-native start --reset-cache` once the terminal display the cache is cleared kill the terminal, after that `npx react-native run-android`

![image](https://user-images.githubusercontent.com/25719095/132120720-b5460c57-fb33-4cdc-a336-e6beaa31031e.png)

i installed "react-native-reanimated". My program always show above error.
